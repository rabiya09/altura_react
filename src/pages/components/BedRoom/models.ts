export interface BedRoomProps {
    title: string;
    description: string;
    readmore: string;
}