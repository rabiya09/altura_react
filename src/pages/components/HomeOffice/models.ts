export interface HomeOfficeProps {
    title: string;
    description: string;
    readmore: string;
}