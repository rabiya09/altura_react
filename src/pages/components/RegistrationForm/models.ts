export interface FieldStructure {
    dirty: boolean; 
    error: boolean; 
    message: string;
}
  
  export interface FieldStr {
    email: FieldStructure;
    password: FieldStructure;
    confirmPassword: FieldStructure;
  }

  export interface SignUpFormState  {
    firstname: string;
    lastname: string;
    email: string;
    password: string
  }
  type Gender = 'male' | 'female' | 'non-binary';
  export interface User {
    name: string;
    age: number;
    email: string;
    gender: Gender;
    password: string;
  }