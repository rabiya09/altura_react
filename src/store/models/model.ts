export interface AddToCartState {
    items: [
        {
            id: 0;
            name: string; 
            description: string; 
            price:  string; 
            imageUrl: string; 
            quantity: number;
        },       
    ];
    addedItems:[
        {
            id: 0;
            name: string; 
            description: string; 
            price:  string; 
            imageUrl: string; 
            qtyInCart: number;
            quantity: number;
        }, 
    ];
    total: 0;   
    invoiceDate: Date;
    invoiceNumber: string; 
  }