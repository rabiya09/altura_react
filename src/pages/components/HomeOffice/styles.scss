.home-office-container {
  width: 100%;
  margin: 0 auto; 
  padding-bottom: 75px;
  float: left;
  @media (max-width: 599px) {
      padding-bottom: 10px;
  }
  .home-office {
      width: 94%;
      margin: 0 auto;
      margin-left: 4%;
      float: left;
      .home-office-text {
          float: left;
          width: 30%;
          margin-top: 175px;
          .read-more {
              width: 100%;
              float: left;
              padding-top: 10px;
              a {
                  color: #E41C77;
                  text-decoration: none;
              }
          }
          @media (max-width: 599px) {
              width: 100%; 
              text-align: center; 
              margin-top: 0px;
          }
      }
      .home-office-title {
          font-weight: bold;
          float: left;
          font-size: 18px;
          width: 100%;
         padding-bottom: 10px;
      }
      .home-office-image {
          float: left;
          width: 58%;
          margin-left: 5%;
          img {
              @media (max-width: 599px) {
                  width: 356px;
                  margin-top: 10px;              
              }
          }
      }
  }
}