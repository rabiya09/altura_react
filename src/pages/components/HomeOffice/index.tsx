import './styles.scss';
import { Link } from "react-router-dom";
import homeoffice from '../../../assets/images/homeoffice.svg';
import { HomeOfficeProps } from './models';

const HomeOffice = (props: HomeOfficeProps) => {
    return (
        <div className="home-office-container">
    <div className="home-office">        
        <div className="home-office-text">
            <span className="home-office-title" data-testid="page-title">{props.title}</span>
            {props.description}
            <span className="read-more"><Link to='/shop-all-home-office'>{props.readmore}</Link></span>
        </div>
        <div className="home-office-image">
            <img src={homeoffice} alt="Home Office"/>
        </div>
    </div>
</div>
    );
}

export default HomeOffice;