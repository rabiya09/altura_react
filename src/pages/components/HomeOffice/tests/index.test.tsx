import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import { getHomeOfficeProps } from './__fakes__';
import { HomeOfficeProps } from '../models';
import HomeOffice from '../index';

describe('HomeOffice', () => {
    const props = getHomeOfficeProps();
    test('should match snapshot', () => {
      const {asFragment} = render(<MemoryRouter><HomeOffice {...getHomeOfficeProps() as HomeOfficeProps}/></MemoryRouter>);
      expect(screen.getByTestId('page-title')).toBeInTheDocument();      
      expect(asFragment()).toMatchSnapshot();
    })  
  });