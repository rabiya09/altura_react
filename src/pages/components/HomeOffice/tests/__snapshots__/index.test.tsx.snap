// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeOffice should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="home-office-container"
  >
    <div
      class="home-office"
    >
      <div
        class="home-office-text"
      >
        <span
          class="home-office-title"
          data-testid="page-title"
        >
          Home Office
        </span>
        Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
        <span
          class="read-more"
        >
          <a
            href="/shop-all-home-office"
          >
            Shop All
          </a>
        </span>
      </div>
      <div
        class="home-office-image"
      >
        <img
          alt="Home Office"
          src="homeoffice.svg"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
